### DISCUSSION TITLE
unpack Problem Firmware - FRITZ!Box 7430 als alien für FRITZ!Box 7412

### DISCUSSION THREAD

**Discussion by aus-sua** [Freetz-NG/freetz-ng#849] [Allgemein] [answered]
Die Erstellung der alien-Firmware einer 7430 für die 7412 läuft sauber durch,
bis es zum "Ent-/Verpacken" des Images kommt.

Die Error:
```
STEP 1A: UNPACK 1ST
source firmware file: dl/fw/FRITZ.Box_7430-07.31.image
checking signature: valid
unpacking firmware image
unpacking outer-filesystem image
 ... ...
unpacking var.tar
detected firmware version: 7430_de-es-fr-it-pl 146.07.31 rev107856 {ALL} [PSQ19P2NL4] (02.09.2023 15:20:51)
done.

ERROR: image could not be verified with public key

STEP 1B: UNPACK 2ND
source firmware file: dl/fw/FRITZ.Box_7430-07.31.image
checking signature: failed
OpenSSL: 40179674F37F0000:error:0200008A:rsa routines:RSA_padding_check_PKCS1_type_1:invalid padding:../crypto/rsa/rsa_pk1.c:75:
40179674F37F0000:error:02000072:rsa routines:rsa_ossl_public_decrypt:padding check failed:../crypto/rsa/rsa_ossl.c:598:
40179674F37F0000:error:1C880004:Provider routines:rsa_verify:RSA lib:../providers/implementations/signature/rsa_sig.c:774:
Verification failure
ERROR: unpacking failed
FINISHED

```

![errors_2x_](https://github.com/Freetz-NG/freetz-ng/assets/73757472/8c09573a-8cb6-40d4-adbb-8386fbf0b95c)

Das sieht für mich "nur" wie ein Pfad- und/oder Variablen-Problem der Firmwaren
aus, die 2. Firmware müsste ja die der 7412 sein!?

freetz: Freetz-NG 22501M-cf7d73b7c master 2023-09-14
Linux: Ubuntu 22.04.3 LTS (als Hyper-V-VM in Win 10)
Den fmake-log und die .config.compressed hänge ich 'mal als zip-Dateien mit an.
[fmake.log_11_+min__2023-09-14___7430-7412_v.07.3x__.zip](https://github.com/Freetz-NG/freetz-ng/files/12640761/fmake.log_11_%2Bmin__2023-09-14___7430-7412_v.07.3x__.zip)
[config.compressed.zip](https://github.com/Freetz-NG/freetz-ng/files/12640762/config.compressed.zip)

PS:
OT
Der "lustige" github-Editor ist einfach nur eine Hipster-Frechheit! :-(

**Comment by fda77** [comment, marked as answer]
Iin deiner config.compressed steht keine firmware .image drin, das macht keinen Sinn.
Juis geht mit Alien nicht da beide images die gleiche juis abfrage nutzen, ich machs einfach aus. Könnte man bestimmt beheben wenn man wollte 

Damit hast du ein falsches image und die signatut passt nicht
```
rm -f dl/fw/FRITZ.Box_7412.137.06.88.image*
```

**Reply by aus-sua** [reply]
> In deiner config.compressed steht keine firmware .image drin...

Aua ..., 
ja, stimmt, sorry.
Ok, also dann doch immer die --->  .config

Danke für die schnelle Änderung.
:-)

Edit:
Sieht gut aus, danke noch einmal. :-)

![i O,_](https://github.com/Freetz-NG/freetz-ng/assets/73757472/7ffd0c28-6fd4-4907-bd13-5c580be20db5)

**Reply by fda77** [reply]
* bfa279fd2
